pkg/metrics/server_test.go

```diff
@@ -19,9 +28,67 @@ var (
 	portOffset uint32 = 0
 )
 
+func TestMain(m *testing.M) {
+	var err error
+
+	mr.Seed(time.Now().UnixNano())
+
+	tlsKey, tlsCRT, err = generateTempCertificates()
+	if err != nil {
+		panic(err)
+	}
+
+	// sets the default http client to skip certificate check.
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{
+		InsecureSkipVerify: true,
+	}
+
+	code := m.Run()
+	os.Remove(tlsKey)
+	os.Remove(tlsCRT)
+	os.Exit(code)
```

Comment:
curious what this achieves?  not apparent to me

if this has some value, perhaps a comment above to explain

Reply:
this is a TestMain method that sets up tlsKey and tlsCert for all the other test methods. We need to do this in order to create server with correct testing tls keys for each test.
TestMain method, if exists, is the main entrypoint for the unit tests. It should return the exit code of `m.Run()`, which runs all the other test methods, and it allows to set common state for the testing environment.
I can add it to the comments if needed.